### PR TITLE
Make OpenLayers 5 available

### PR DESCRIPTION
**Pull request by mvdstruijk**

### CODE REVIEW

File: viewer/src/main/webapp/WEB-INF/jsp/app.jsp [hunk 1]
```diff
@@ -99,8 +99,23 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
                 </c:otherwise>
             </c:choose>
         </c:if>
-        <script type="text/javascript" src="${contextPath}/viewer-html/common/overrides.js"></script>
-        <script type="text/javascript" src="${contextPath}/resources/js/FileSaver.js"></script>
```
Comment (mprins) [inline viewer/src/main/webapp/WEB-INF/jsp/app.jsp:103]: Why are these removed? and if these are no longer necessary - which I doubt- why aren't these files deleted?

File: viewer/src/main/webapp/WEB-INF/jsp/app.jsp [hunk 2]
```diff
@@ -121,7 +136,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
                 <script type="text/javascript" src="${contextPath}/viewer-html/components/RequestManager.js"></script>
                 <script type="text/javascript" src="${contextPath}/viewer-html/components/DataSelectionChecker.js"></script>
 
-                <script type="text/javascript" src="${contextPath}/viewer-html/common/overrides.js"></script>
```
Comment (mprins) [inline viewer/src/main/webapp/WEB-INF/jsp/app.jsp:124]: sameasabove

File: viewer/pom.xml [hunk 3]
```diff
@@ -296,7 +296,7 @@
                                 <param>${viewerhtml.path}/components/RequestManager.js</param>
                                 <param>${viewerhtml.path}/components/DataSelectionChecker.js</param>
                                 <!-- Common -->
-                                <param>${viewerhtml.path}/common/overrides.js</param>
```
Comment (mprins) [inline viewer/pom.xml:311]: if we don't need this file it should be deleted from the repo

File: viewer/src/main/webapp/WEB-INF/jsp/app.jsp [hunk 4]
```diff
@@ -99,8 +99,23 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
                 </c:otherwise>
             </c:choose>
         </c:if>
-        <script type="text/javascript" src="${contextPath}/viewer-html/common/overrides.js"></script>
-        <script type="text/javascript" src="${contextPath}/resources/js/FileSaver.js"></script>
+                  
+        <c:if test="${actionBean.viewerType == 'ol'}">
+               <link href="${contextPath}/viewer-html/common/ol/ol.css" rel="stylesheet">  
+               <link href="${contextPath}/viewer-html/common/resources/css/openlayers.css" rel="stylesheet">
+               <script src="http://cdnjs.cloudflare.com/ajax/libs/proj4js/2.3.6/proj4.js" type="text/javascript"></script>
```
Comment (mprins) [inline viewer/src/main/webapp/WEB-INF/jsp/app.jsp]: this should be using https otherwise loading will fail on https sites (using https will not fail on http sites). maybe better to put this file in the repo so it is available locally

File: viewer/src/main/webapp/viewer-html/common/viewercontroller/openlayers/OpenLayersMeasure.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+/* 
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (mprins) [inline viewer/src/main/webapp/viewer-html/common/viewercontroller/openlayers/OpenLayersMeasure.js]: should use proper license or nothing

File: viewer/src/main/webapp/viewer-html/common/viewercontroller/ol/tools/ZoomOutButton.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+/* 
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (mprins) [inline viewer/src/main/webapp/viewer-html/common/viewercontroller/ol/tools/ZoomOutButton.js:16]: should use proper license or nothing

**Review by mprins — CHANGES_REQUESTED** [review]
not run the code, but:
- I have questions about removing `overrides.js`
- Most /all of the new files don't have a proper license header, the should have a proper license or nothing, but not whatever netbeans decided to throw in.

File: viewer/src/main/webapp/viewer-html/common/viewercontroller/ViewerController.js [hunk 7]
```diff
@@ -145,6 +145,9 @@ Ext.define("viewer.viewercontroller.ViewerController", {
             this.mapComponent = new viewer.viewercontroller.FlamingoMapComponent(this, mapId,config);
         }else if(viewerType === "openlayers") {
             this.mapComponent = new viewer.viewercontroller.OpenLayersMapComponent(this, mapId,config);
+        }else if(viewerType === "ol"){
```
Comment (geertplaisier) [inline viewer/src/main/webapp/viewer-html/common/viewercontroller/ViewerController.js]: Do not really like the "ol" viewerType. This will be very difficult to find / replace. Why not "openlayers5" for example? Easier to find, easier to replace

**Comment by mprins** [comment]
@mvdstruijk Something went wrong; javascript compiler error (maybe missng a comma in OpenLayersMapComponent.js) :

```
SEVERE: /home/travis/build/flamingo-geocms/flamingo/viewer/src/main/webapp/viewer-html/common/viewercontroller/openlayers/OpenLayersMapComponent.js:842: ERROR - Parse error. '}' expected
    deselectAllOtherFeatures: function(layer){
    ^
Sep 06, 2019 8:28:48 AM com.google.javascript.jscomp.LoggerErrorManager printSummary
WARNING: 1 error(s), 0 warning(s)
[ERROR] JSC_PARSE_ERROR. Parse error. '}' expected at /home/travis/build/flamingo-geocms/flamingo/viewer/src/main/webapp/viewer-html/common/viewercontroller/openlayers/OpenLayersMapComponent.js line 842 : 4
```

 see: https://travis-ci.org/flamingo-geocms/flamingo/jobs/581561832#L1610-L1616

**Comment by mtoonen** [comment]
- [ ] WMTS gives https://openlayers.org/en/v5.3.0/doc/errors/#18 error
- [ ] google navigation fails on line https://github.com/flamingo-geocms/flamingo/blob/impr/openlayers5/viewer/src/main/webapp/viewer-html/common/viewercontroller/ol/OlTool.js#L33
- [ ]

**Comment by mtoonen** [comment]
- [x] TypeError: Cannot read property 'toLowerCase' of undefined at constructor.setTool (OlTool.js:34)
- [x] WMTSgives error: "Assertion failed. See https://openlayers.org/en/v5.3.0/doc/errors/#18 for details."

File: viewer/src/main/webapp/WEB-INF/jsp/app.jsp [hunk 8]
```diff
@@ -98,9 +98,26 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
             </c:choose>
+            <script type="text/javascript" src="${contextPath}/viewer-html/common/overrides.js"></script>
         </c:if>
-        <script type="text/javascript" src="${contextPath}/viewer-html/common/overrides.js"></script>
         <script type="text/javascript" src="${contextPath}/resources/js/FileSaver.js"></script>
+            
+        <c:if test="${actionBean.viewerType == 'openlayers5'}">
+               <link href="${contextPath}/viewer-html/common/ol/ol.css" rel="stylesheet">  
+               <link href="${contextPath}/viewer-html/common/resources/css/openlayers.css" rel="stylesheet">
+               <script src="https://cdnjs.cloudflare.com/ajax/libs/proj4js/2.3.6/proj4.js" type="text/javascript"></script>
```
Comment (geertplaisier) [inline viewer/src/main/webapp/WEB-INF/jsp/app.jsp]: I think its better to include this JS in our own source instead of depending on an external service

**Comment by mtoonen** [comment]
Blocking:
- [ ] snapping in combination with measuring

**Review by mtoonen — CHANGES_REQUESTED** [review]
Blocking:
- [x] Snapping in combination with measuring

@mtoonen Snapping in combination with measuring is now also working with OL5

File: viewer/src/main/webapp/viewer-html/common/viewercontroller/ol/tools/ZoomOutButton.js (hunk 6 above)
Comment (mvdstruijk) [reply viewer/src/main/webapp/viewer-html/common/viewercontroller/ol/tools/ZoomOutButton.js:16]: Done

File: viewer/src/main/webapp/viewer-html/common/viewercontroller/openlayers/OpenLayersMeasure.js (hunk 5 above)
Comment (mvdstruijk) [reply viewer/src/main/webapp/viewer-html/common/viewercontroller/openlayers/OpenLayersMeasure.js]: Done

File: viewer/src/main/webapp/WEB-INF/jsp/app.jsp (hunk 1 above)
Comment (mvdstruijk) [reply viewer/src/main/webapp/WEB-INF/jsp/app.jsp:103]: Overrides.js is now only included when viewertype == OpenLayers  aka ol2
Filesaver.js was a mistake.

File: viewer/src/main/webapp/WEB-INF/jsp/app.jsp (hunk 2 above)
Comment (mvdstruijk) [reply viewer/src/main/webapp/WEB-INF/jsp/app.jsp:124]: Moved to OpenLayers 2 includes.

File: viewer/pom.xml (hunk 3 above)
Comment (mvdstruijk) [reply viewer/pom.xml:311]: Moved to OpenLayers 2 includes.

**Review by mtoonen — APPROVED** [review]
LGTM
